shell/src-app/Linter.hs

```diff
@@ -0,0 +1,83 @@
+-- Allow ourselves to write Text literals.
+{-# LANGUAGE OverloadedStrings #-}
+
+module Main where
+
+    import Prologue hiding (Symbol)
+    import Luna.Syntax.Text.Lexer.Runner
+    import Luna.Syntax.Text.Lexer.Token as T
+    import Luna.Syntax.Text.Lexer.Symbol as S
+    import Data.Text.Position            (Delta)
+    import Data.Text32 hiding (convert)
+    import Data.Aeson
+    import qualified Data.ByteString.Lazy as B
+
+    instance ToJSON Delta where
+        toJSON delta = toJSON ((convert delta) :: Int)
+
+    jsonStr :: String -> Value
+    jsonStr = toJSON
+
+    jsonT32 :: Text32 -> Value
+    jsonT32 txt = toJSON ((convert txt) ::String)
```

Comment:
Dont use so short names. It is just not understandable. If you really want to create such alias name it `text32ToJSON`

Reply:
Moreover, it does not make sense to make an util function for that. Just implement `instance ToJSON Text32`. This typeclass is for exactly this abstraction.

Reply:
The issue is, when I implement that instance I get
```
        • Overlapping instances for ToJSON Text32
            arising from a use of ‘toJSON’
          Matching instances:
            instance (vector-0.12.0.1:Data.Vector.Generic.Base.Vector Vector a,
                      ToJSON a) =>
                     ToJSON (Vector a)
              -- Defined in ‘aeson-1.3.1.1:Data.Aeson.Types.ToJSON’
            instance ToJSON Text32 -- Defined at src-app/Linter.hs:18:14
        • In the second argument of ‘($)’, namely ‘toJSON n’
          In the expression: Just $ toJSON n
          In an equation for ‘valueForSymbol’:
              valueForSymbol (Var n) = Just $ toJSON n
       |
    29 |     valueForSymbol (Var n) = Just $ toJSON n
       |                                     ^^^^^^^^
    
```
because there's an instance ToJSON (Vector a), but it's behaviour is not what I want (it encodes "abc" as ["a","b","c"].
I don't know how to hide it.

Reply:
In this case you should use `instance {-# OVERLAPPING #-} ToJSON Text32`. It should land in the Text32 implementation later, but for now just use it locally